shared/src/types/keccak.rs

```diff
@@ -0,0 +1,215 @@
+//! This module is for hashing Anoma types using the keccak
+//! hash function in a way that is compatible with smart contracts
+//! on Ethereum
+use std::convert::TryFrom;
+use std::fmt::Display;
+
+use borsh::{BorshDeserialize, BorshSchema, BorshSerialize};
+use hex::FromHex;
+use thiserror::Error;
+
+use crate::types::hash::{Hash, HASH_LENGTH};
+
+#[allow(missing_docs)]
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("TEMPORARY error: {error}")]
+    Temporary { error: String },
+    #[error("Failed trying to convert slice to a hash: {0}")]
+    ConversionFailed(std::array::TryFromSliceError),
+    #[error("Failed to convert string into a hash: {0}")]
+    FromStringError(hex::FromHexError),
+}
+
+/// A Keccak hash
```

Comment:
```suggestion
/// Represents a Keccak hash.
```
not a fan of docstrings starting with articles